### PR TITLE
luci-base: network.js: Show clients also from wifi VLANs.

### PR DESCRIPTION
**Pull request by oldium**
Iterate through VLANs to get the clients connected on configured wifi
VLANs (as configured by `wifi-vlan` sections in `wireless` configuration).

This is a minimum support for VLANs on wireless network. The VLAN name is
nowhere displayed, but at least clients using VLANs are visible.

### CODE REVIEW

File: modules/luci-base/htdocs/luci-static/resources/network.js [hunk 1]
```diff
@@ -3570,6 +3570,28 @@ WifiNetwork = baseclass.extend(/** @lends LuCI.network.WifiNetwork.prototype */
...
+	 * @returns {string[]}
+	 * Returns the current Linux VLAN network device name as resolved
+	 * from `ubus` runtime information or empty array if this network
+	 * has no associated VLAN network devices.
+	 */
+	getVlanIfnames: function() {
+		var vlans = this.ubus('net', 'vlans'),
+		    ifnames = [];
+
+		if (vlans == null)
```
Comment (jow-) [inline modules/luci-base/htdocs/luci-static/resources/network.js:3585]: ... you can then subsequently drop this check and simply loop over the empty array below...

File: modules/luci-base/htdocs/luci-static/resources/network.js [hunk 2]
```diff
@@ -3570,6 +3570,28 @@ WifiNetwork = baseclass.extend(/** @lends LuCI.network.WifiNetwork.prototype */
...
+			return ifnames;
+
+		for (var i = 0; i < vlans.length; i++) {
```
Comment (jow-) [inline modules/luci-base/htdocs/luci-static/resources/network.js:3588]: Please leave out the curly braces for single-statement loops.

File: modules/luci-base/htdocs/luci-static/resources/network.js [hunk 3]
```diff
@@ -3880,7 +3902,17 @@ WifiNetwork = baseclass.extend(/** @lends LuCI.network.WifiNetwork.prototype */
 	 * with this network.
 	 */
 	getAssocList: function() {
-		return callIwinfoAssoclist(this.getIfname());
+		var tasks = [];
+		var ifnames = Array.prototype.concat.apply(
```
Comment (jow-) [inline modules/luci-base/htdocs/luci-static/resources/network.js:3906]: I'd prefer this simpler expression: `var ifnames = [ this.getIfname() ].concat(this.getVlanIfnames());`

File: modules/luci-base/htdocs/luci-static/resources/network.js [hunk 4]
```diff
@@ -3880,7 +3902,17 @@ WifiNetwork = baseclass.extend(/** @lends LuCI.network.WifiNetwork.prototype */
...
+			[this.getIfname()], this.getVlanIfnames());
+
+		for (var i = 0; i < ifnames.length; i++) {
```
Comment (jow-) [inline modules/luci-base/htdocs/luci-static/resources/network.js:3909]: Please leave out the curly braces for single-statement loops.

**Review by jow- — CHANGES_REQUESTED** [review]

**Comment by jow-** [comment]
Thanks a lot for this change! I didn't know that the AP/VLANs are exposed via ubus nowadays, this simplifies a lot. I added a couple of minor change requests, if those are addressed, the code looks good to merge.

**Comment by oldium** [comment]
Done. If you wish, I can squash it into 1 commit.

**Comment by jow-** [comment]
Yes, please squash them :)

**Comment by oldium** [comment]
Done

**Comment by jow-** [comment]
Merged, thanks!

**Pull request merged**